pydicom/filereader.py

```diff
@@ -282,6 +294,12 @@ def data_element_generator(fp,
         if length != 0xFFFFFFFF:
             # don't defer loading of Specific Character Set value as it is
             # needed immediately to get the character encoding for other tags
+            if tag_set is not None and tag not in tag_set and tag != (
```

Comment:
this is complicated to read. I think this line does the same and its easier to read:
```py
if tag not in tag_set.update([0x08, 0x05]):
```